File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-2.java
class Solution {
    public void setZeroes(int[][] matrix) {
        boolean zeroInRow = false;
        boolean zeroInCol = false;
        int row = matrix.length;
        int col = matrix[0].length;
        
        for (int i = 0; i < row; i++) {
            if (matrix[i][0] == 0) {
                zeroInRow = true;
                break;
            }
        }

        for (int j = 0; j < col; j++) {
            if (matrix[0][j] == 0) {
                zeroInCol = true;
                break;
            }
        }

        for (int i = 1; i < row; i++) {
            for (int j = 1; j < col; j++) {
                if (matrix[i][j] == 0) {
                    matrix[i][0] = 0;
                    matrix[0][j] = 0;
                }
            }
        }

        for (int i = 1; i < row; i++) {
            if (matrix[i][0] == 0) {
                for (int j = 1; j < col; j++) matrix[i][j] = 0;
            }
        }

        for (int j = 1; j < col; j++) {
            if (matrix[0][j] == 0) {
                for (int i = 1; i < row; i++) matrix[i][j] = 0;
            }
        }

        if (zeroInRow) {
            for (int i = 0; i < row; i++) matrix[i][0] = 0;
        }

        if (zeroInCol) {
            for (int j = 0; j < col; j++) matrix[0][j] = 0;
        }

        return;
    }
}

File: Data Structures & Algorithms/search-for-word/submission-2.py
class Solution:
    def exist(self, board: List[List[str]], word: str) -> bool:
        
        def dfs(i: int, row: int, col: int) -> bool:
            if i == len(word):
                return True
            if not (0 <= row < len(board) and 0 <= col < len(board[0])) or board[row][col] != word[i]:
                return False
            
            temp = board[row][col]
            board[row][col] = '.'
            for nx, ny in [(0, 1), (1, 0), (-1, 0), (0, -1)]:
                next_row = nx + row
                next_col = ny + col
                if dfs(i + 1, next_row, next_col):
                    return True
            board[row][col] = temp
            return False
        
        for i in range(len(board)):
            for j in range(len(board[0])):
                if dfs(0, i, j):
                    return True
        return False
